backend/internal/system/database/client/dbclient.go

```diff
@@ -92,6 +95,39 @@ func (client *DBClient) ExecuteQuery(query model.DBQuery, args ...interface{}) (
 	return results, nil
 }
 
+// executeNonQuery executes an INSERT, UPDATE, or DELETE query and returns the number of rows affected.
+func (client *DBClient) executeNonQuery(opType, queryID, sql string, args ...interface{}) (int64, error) {
```

Comment:
```suggestion
func (client *DBClient) executeNonQuery(opType, query model.DBQuery, args ...interface{}) (int64, error) {
```
What if we pass the entire query struct to this method as well?

Reply:
Fixed with: https://github.com/asgardeo/thunder/compare/ef36cc78465dc18fdc6e4d2e7a3fb54d8604b067..56b8ddcc66ae1cec9a1ea8872683128bb8bd2111